islandora_premis.module

```diff
@@ -81,11 +81,19 @@ function islandora_premis_xml_download($output) {
  *   The content to download.
  */
 function _islandora_premis_xml_download($output) {
-  drupal_add_http_header('Content-type', 'text/xml;charset=utf8');
-  drupal_add_http_header('Content-length', strlen($output));
-  drupal_add_http_header('Content-Disposition', 'attachment; filename="premis.xml"');
-  print $output;
-  drupal_page_footer();
+  $pid = arg(2);
```

Comment:
Nice Mark, thanks. Quick question (on a train, can't test so please excuse me). This arg(2) piece, how does it behave when the wrapper is used to call this?  ``function islandora_premis_xml_download($output) {``
this one https://github.com/Islandora/islandora_premis/blob/1f8614b18a9e9e025a0378f270727df40030b27b/islandora_premis.module#L73 is there a second argument in that case?

Reply:
Good question @DiegoPino, but if anyone is calling that function, they already have the PREMIS XML (it's in `$output`), presumably by calling `islandora_premis_transform_foxml_to_premis($islandora_object)`. So I'm not sure why they would do `islandora_premis_xml_download($output)`.

Reply:
nice! So seems like since `[arg()](https://api.drupal.org/api/drupal/includes%21bootstrap.inc/function/arg/7.x)` depends on the context of the currently loaded page, if this function is always used in the same path /route/page context we should be ok! good.  Now i have another question: Does`[menu_get_object()](https://api.drupal.org/api/drupal/includes%21menu.inc/function/menu_get_object/7.x)` behave differently than the combination of $pid and islandora _load()? thanks 😁

Reply:
I just did a smoketest with `menu_get_object()` and it appears to work the same as `islandora_object_load()`. If you're recommending that I use it instead of `islandora_object_load($pid)`, I'm happy to.

Reply:
Thanks a lot, @mjordan. As always, Drupal documentation is an exploratory trip into the unknown for me so just read a bit more.  The  arg() one says the functions is not recommended because of its legibility. Also some other modules have moved to menu_get_object and get_item, But again, no expert. Google says it can maybe be more efficient to do menu_get_object()? in the context of a correct path, it will indeed do the argument extraction and the object load combined. If it's not too much to ask would like to try menu_get_object() here. Thanks a lot again for your work on this and disposition to read all my requests.

Reply:
OK, `menu_get_object()` is now in place. Thanks for the suggestion.